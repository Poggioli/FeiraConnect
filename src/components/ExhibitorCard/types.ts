import { GetExhibitorsByStreetMarket } from "@/services/getExhibitorsByStreetMarket";

export type ExhibitorCardProps = GetExhibitorsByStreetMarket;

export type UseExhibitorCardPropsProps = Pick<ExhibitorCardProps, "name" | "phone" | "whatsApp" | "website">;
