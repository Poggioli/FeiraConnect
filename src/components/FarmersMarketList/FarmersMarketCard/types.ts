import { Frequency, SearchFarmersMarketByCityId } from "@/services/searchFarmersMarketByCityId";

export type FarmersMarketCardProps = SearchFarmersMarketByCityId

export type UseFarmersMarketCardProps = {
  frequency: Frequency,
  startTime: Date,
  endTime: Date
}