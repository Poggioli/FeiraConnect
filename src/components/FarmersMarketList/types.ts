export type FarmersMarketListProps = {
  city: string;
};

export type UseFarmersMarketListProps = {
  city: string
}