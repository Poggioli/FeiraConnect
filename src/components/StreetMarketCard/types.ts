import { GetStreetMarketsByCity } from "@/services/getStreetMarketsByCity";

export type StreetMarketCardProps = GetStreetMarketsByCity;

export type UseStreetMarketCardProps = Pick<
  StreetMarketCardProps,
  "name" | "neighborhood" | "weekday" | "apperture" | "closure"
>;
