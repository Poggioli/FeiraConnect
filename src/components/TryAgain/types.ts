export type TryAgainProps = {
  refetch: () => unknown;
}