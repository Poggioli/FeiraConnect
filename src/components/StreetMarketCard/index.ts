export * from "./StreetMarketCard";
export * from "./types";
export * from "./useStreetMarketCard";
