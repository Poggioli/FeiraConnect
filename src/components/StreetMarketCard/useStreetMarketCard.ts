import { Weekday } from "@/services/getStreetMarketsByCity";
import { UseStreetMarketCardProps } from "./types";

const weekdaysFormated: Record<Weekday, string> = {
  saturday: "Sábado",
  sunday: "Domingo",
  monday: "Segunda-feira",
  tuesday: "Terça-feira",
  wednesday: "Quarta-feira",
  thursday: "Qinita-feira",
  friday: "Sexta-feira",
};

export function useStreetMarketCard({
  apperture,
  closure,
  name,
  neighborhood,
  weekday,
}: UseStreetMarketCardProps) {
  function formatHour(hour: number): string {
    return String(hour).padStart(2, "0");
  }

  const formatedName = name.charAt(0).toUpperCase() + name.slice(1);

  const formatedNeighborhood =
    neighborhood.charAt(0).toUpperCase() + neighborhood.slice(1);

  const timeFormated = `${formatHour(apperture)}h - ${formatHour(closure)}h`;

  const weekdayFormated = weekdaysFormated[weekday];

  return {
    formatedName,
    formatedNeighborhood,
    timeFormated,
    weekdayFormated,
  };
}
