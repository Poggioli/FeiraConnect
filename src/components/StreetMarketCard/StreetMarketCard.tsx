import { Badge } from "@/components/ui/badge";
import { Card, CardDescription, CardTitle } from "@/components/ui/card";
import { cn } from "@/lib/utils";
import { FC, HTMLAttributes, forwardRef } from "react";
import { StreetMarketCardProps } from "./types";
import { useStreetMarketCard } from "./useStreetMarketCard";
import { Skeleton } from "../ui/skeleton";

export const StreetMarketCard = forwardRef<
  HTMLDivElement,
  HTMLAttributes<HTMLDivElement> & StreetMarketCardProps
>(({
  name,
  neighborhood,
  apperture,
  closure,
  weekday,
  ...props
}, ref) => {
  const {
    formatedName,
    formatedNeighborhood,
    timeFormated,
    weekdayFormated
  } = useStreetMarketCard({
    apperture,
    closure,
    name,
    neighborhood,
    weekday,
  });

  return (
    <Card {...props} className={cn("py-4 px-5 w-full", props.className)} ref={ref}>
      <CardTitle>{formatedName}</CardTitle>
      <CardDescription>{formatedNeighborhood}</CardDescription>
      <div className="mt-2 flex flex-row wrap gap-2">
        <Badge variant="outline">{weekdayFormated}</Badge>
        <Badge variant="outline">{timeFormated}</Badge>
      </div>
    </Card>
  );
});

export const StreetMarketCardSkeleton: FC = () => (
  <Card className="py-4 px-5 w-full">
    <div className="flex flex-col justify-center items-start gap-1">
      <Skeleton className="rounded-sm h-4 w-36" />
      <Skeleton className="rounded-sm h-[0.875rem] w-14" />
    </div>
    <div className="mt-2 flex flex-row wrap gap-2">
      <Skeleton className="rounded-sm h-4 w-28" />
      <Skeleton className="rounded-sm h-4 w-20" />
    </div>
  </Card>
)
