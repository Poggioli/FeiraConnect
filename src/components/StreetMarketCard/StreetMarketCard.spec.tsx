import { customRender } from "@tests/customRender";
import { StreetMarketCard, StreetMarketCardSkeleton } from "./StreetMarketCard";
import { StreetMarketCardProps } from "./types";
import { waitFor } from "@testing-library/dom";
import { render } from "@testing-library/react";

describe('@/components/StreetMarketCard', () => {

  const defaultValue: StreetMarketCardProps = {
    id: '1',
    slug: 'name-name',
    apperture: 9,
    closure: 15,
    name: "name name",
    neighborhood: "neighborhood",
    weekday: "saturday",
    location: ''
  };

  it(`GIVEN a StreetMarketCard component
      WHEN render
      THEN should match with snapshot`, async () => {
    const { baseElement, getByText } = customRender(<StreetMarketCard {...defaultValue} />);

    await waitFor(() => {
      const streetMarketName = getByText("Name name");
      expect(streetMarketName).toBeVisible();
      expect(baseElement).toMatchSnapshot();
    })
  });

  it(`GIVEN a StreetMarketCardSkeleton component
      WHEN render
      THEN should match with snapshot`, () => {
    const { baseElement } = render(<StreetMarketCardSkeleton />);
    expect(baseElement).toMatchSnapshot();
  });
});