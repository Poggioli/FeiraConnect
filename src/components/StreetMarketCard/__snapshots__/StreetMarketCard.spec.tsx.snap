// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@/components/StreetMarketCard > GIVEN a StreetMarketCard component
      WHEN render
      THEN should match with snapshot 1`] = `
<body>
  <div>
    <a
      from="/city/$city"
      href="/city/undefined/name-name"
    >
      <div
        class="rounded-xl border bg-card text-card-foreground py-4 px-5 w-full"
        id="1"
        location=""
      >
        <h3
          class="font-semibold leading-none tracking-tight"
        >
          Name name
        </h3>
        <p
          class="text-sm text-muted-foreground"
        >
          Neighborhood
        </p>
        <div
          class="mt-2 flex flex-row flex-wrap gap-2"
        >
          <div
            class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground"
          >
            SÃ¡bado
          </div>
          <div
            class="inline-flex items-center rounded-md border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground"
          >
            09h - 15h
          </div>
        </div>
      </div>
    </a>
  </div>
</body>
`;

exports[`@/components/StreetMarketCard > GIVEN a StreetMarketCardSkeleton component
      WHEN render
      THEN should match with snapshot 1`] = `
<body>
  <div>
    <div
      class="rounded-xl border bg-card text-card-foreground py-4 px-5 w-full"
    >
      <div
        class="flex flex-col justify-center items-start gap-1"
      >
        <div
          class="animate-pulse bg-primary/10 rounded-sm h-4 w-36"
        />
        <div
          class="animate-pulse bg-primary/10 rounded-sm h-[0.875rem] w-14"
        />
      </div>
      <div
        class="mt-2 flex flex-row flex-wrap gap-2"
      >
        <div
          class="animate-pulse bg-primary/10 rounded-sm h-4 w-28"
        />
        <div
          class="animate-pulse bg-primary/10 rounded-sm h-4 w-20"
        />
      </div>
    </div>
  </div>
</body>
`;
